docs/modules/pixbuf.md

```diff
@@ -0,0 +1,274 @@
+# Pixel Buffer (pixbuf) Module
+
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2020-?? | [nwf](https://github.com/nwf) | nwf | [pixbuf.c](../../app/modules/pixbuf.c) |
+
+The pixbuf library offers C-array byte objects and convenient utility functions
+for maintaining small frame buffers, usually for use with LED arrays, as
+supported by, e.g., ws2812.
+
+## pixbuf.newBuffer()
+Allocate a new memory buffer to store LED values.
+
+#### Syntax
+`pixbuf.newBuffer(numberOfLeds, type)`
+
+#### Parameters
+ - `numberOfLeds` length of the LED strip
+ - `type` the type of the LED srip.  Must be one of
+
+    - `pixbuf.TYPE_RGB` for Red-Green-Blue arrays
+    - `pixbuf.TYPE_GRB` for Green-Red-Blue arrays
+    - `pixbuf.TYPE_RGBW` or `pixbuf.TYPE_GRBW` for the above but also with a White channel
+    - `pixbuf.TYPE_WWA` for warm-White cool-White Amber arrays
+    - `pixbuf.TYPE_IBGR` for Intensity-mediated Blue-Green-Red arrays
+
+#### Returns
+`pixbuf.buffer` object
+
+## pixbuf.buffer:get()
+Return the value at the given position
+
+#### Syntax
+`buffer:get(index)`
+
+#### Parameters
+ - `index` position in the buffer (1 for first LED)
+
+#### Returns
+`(color)`
+
+#### Example
+```lua
+buffer = pixbuf.newBuffer(32, pixbuf.TYPE_RGBW)
+print(buffer:get(1))
+0	0	0	0
+```
+
+## pixbuf.buffer:set()
+Set the value at the given position
+
+#### Syntax
+`buffer:set(index, color)`
+
```

Comment:
As mentioned above, the goal should be to have the same user code for RGB and GRB buffers, so we need to swat them either here or on output. (or in a later PR?)

Reply:
Done.